src/js/components/navbars.js

```diff
@@ -85,48 +86,127 @@
-   *  <div class="navbar navbar-app navbar-absolute-bottom">
-   *    <!-- ... -->
-   *  </div>
-   * ```
-   */
+  * @directive navController
+  * @restrict C
+  * @description
+  *
+  * Directive to bind controller to navbar
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-top' nav-controller>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
+  module.directive('navController', function() {
+    return {
+      restrict: 'A',
+      controller: function() {
+        this.registerController = function(name, ctrl) {
+          this[name] = ctrl;
+        };
+      },
+      link: function(scope) {}
+    };
+  });
+
+ /**
+  * @directive navbarAbsoluteTop
+  * @restrict C
+  * @description
+  *
+  * Setup absolute positioned top navbar.
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-top'>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
+
+ /**
+  * @directive navbarAbsoluteBottom
+  * @restrict C
+  * @description
+  *
+  * Setup absolute positioned bottom navbar.
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-bottom'>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
   angular.forEach(['top', 'bottom'], function(side) {
     var directiveName = 'navbarAbsolute' + side.charAt(0).toUpperCase() + side.slice(1);
     module.directive(directiveName, [
       '$rootElement',
       function($rootElement) {
         return {
           restrict: 'C',
-          link: function(scope) {
+          require: 'navController',
+          link: function(scope, elem, attr, ctrl) {
             $rootElement.addClass('has-navbar-' + side);
             scope.$on('$destroy', function() {
               $rootElement.removeClass('has-navbar-' + side);
             });
+            if (!global) global = ctrl;
+            global.registerController(side, new (function() {
+              this.hide = function() {
+                $rootElement.removeClass('has-navbar-' + side);
+                elem.addClass('ng-hide');
+              };
+              this.show = function() {
+                elem.removeClass('ng-hide');
+                $rootElement.addClass('has-navbar-' + side);
+              };
+              this.slideout = function() {
+                $rootElement.removeClass('has-navbar-' + side);
+                elem.addClass('ng-hide');
+              };
+              this.slidein = function() {
+                elem.removeClass('ng-hide');
+                $rootElement.addClass('has-navbar-' + side);
+              };
+            })());
           }
         };
       }
     ]);
   });
+
+  /**
+   * @directive [hide|show|slidein|slideout][Top|Bottom|Both][Nav]
+   * @restrict C
+   * @description
+   *
+   * Toggle each(or both) nabar(s) with or without animations
+   *
+   * ``` html
+   *  <view class='[hide|show|slidein|slideout]-[top|bottom|both]-[nav]'>
+   *    <!-- ... -->
+   *  </view>
+   * ```
+   */
+  angular.forEach(['Top', 'Bottom', 'Both'], function(nav) {
```

Comment:
Why is this a directive? I mean, i kinda got that you are invoking the controller while linking on a view element, so perhaps you like to react to page changes (`ng-view`?), so when you navigate on some areas of your app you don't want to see the nav, while some other has it.

In that case, but also in general i suggest to re-approach the problem splitting it into 2 distinct phases:

- 1) easily allow to hide, show the navbars (the slide part is unnecessary cause is better and easier achieved and customized via css)
- 2) provide an easy way (if that's what you wanna achieve) to hide/show the navbar on certain pages (or when you navigate somewhere, or just the first time you open the app... let's say ..bound to the location)

Currently what you achieved with 2 directives could be done with a few css classes. Let's discuss outside the PR.